### PR TITLE
112 frontend add invite to games

### PR DESCRIPTION
**Pull request by tsiguenz**
Logic is working but front is bad.

To improve:
- do the front obviously
- can create message type where we can click on it to avoid copy past

### CODE REVIEW

File: backend/src/chat/chat.gateway.ts [hunk 1]
```diff
@@ -351,6 +350,54 @@ export class ChatGateway
...
+        chatroom: chatroom
+      });
+      const dest = await this.users.getUserById(payload.destId);
+      const destSocket = await this.getUserSocket(dest.id);
+      if (destSocket) {
+        destSocket.emit(events.CHATROOM_NEW, {
+          chatroom: chatroom
+        });
+      }
+      client.disconnect();
```
Comment (gmorange42) [inline backend/src/chat/chat.gateway.ts:394]: Is it good to in chat view ?

File: backend/src/chat/chat.gateway.ts [hunk 2]
```diff
@@ -351,6 +350,54 @@ export class ChatGateway
...
   }
 
+  @SubscribeMessage(events.GAME_INVITATION)
+  async handleGameInvitation(
+    @ConnectedSocket() client: Socket,
+    @MessageBody() payload: { gameUrl: string; destId: string }
+  ) {
+    try {
+      if (!client['decoded'] || client['decoded'].sub === payload.destId)
+        client.disconnect();
```
Comment (Crysicia) [inline backend/src/chat/chat.gateway.ts:360]: Why would you disconnect the user if it's the destId ?

File: backend/src/chat/chat.gateway.ts [hunk 3]
```diff
@@ -351,6 +350,54 @@ export class ChatGateway
...
+      const user = await this.users.getUserById(client['decoded'].sub);
+      const chatroom = await this.privateMessage.findOrCreate(
+        user.id,
+        payload.destId
+      );
+      if (
+        await this.chatroomRestriction.isUserRestricted(user.id, chatroom.id)
+      ) {
+        client.disconnect();
```
[the remark this answers is not in the document]
Comment (Crysicia) [reply backend/src/chat/chat.gateway.ts:369]: Also, there cannot be any restriction on a one_to_one room

File: backend/src/chat/chat.gateway.ts [hunk 4]
```diff
@@ -351,6 +350,54 @@ export class ChatGateway
...
+        payload.gameUrl
+      );
+      this.server.to(chatroom.id).emit(events.MESSAGE_TO_CLIENT, {
+        authorId: user.id,
+        authorNickname: user.nickname,
+        authorAvatarUrl: user.avatarPath,
+        chatroomId: chatroom.id,
+        sentAt: message.createdAt,
+        data: payload.gameUrl
+      });
```
Comment (Crysicia) [inline backend/src/chat/chat.gateway.ts:383]: This could be extracted to it's own function now

**Review by gmorange42 — COMMENTED** [review]
Nice job !

**Review by Crysicia — CHANGES_REQUESTED** [review]
Why do you disconnect the user after sending the invitation?
Doesn't it break the chat?

**Pull request merged**